src/uu/seq/src/floatparse.rs

```diff
@@ -0,0 +1,230 @@
+    let value = parse_float(s)?;
+    let number = BigDecimal::from_f64(value).ok_or(ParseNumberError::Float)?;
+    let fractional_digits = i64::max(number.fractional_digit_count(), 0) as usize;
+    Ok(PreciseNumber::new(
+        ExtendedBigDecimal::BigDecimal(number),
+        0,
+        fractional_digits,
+    ))
+}
+
+/// Parse a floating-point number from a hexadecimal notation.
+///
+/// # Errors
+///
+/// This function returns an error if:
+/// - the input string is not a hexadecimal string
+/// - input data can't be interpreted as ['f64'] and ['BigDecimal']
+fn parse_float(s: &str) -> Result<f64, ParseNumberError> {
+    let mut s = s.trim();
+
+    // Detect a sign
+    let sign = if s.starts_with('-') {
+        s = &s[1..];
+        -1.0
+    } else if s.starts_with('+') {
+        s = &s[1..];
+        1.0
+    } else {
+        1.0
+    };
+
+    // Return error if not a Hex string
+    if !s.starts_with("0x") && !s.starts_with("0X") {
+        return Err(ParseNumberError::Float);
+    }
+
+    // Skip Hex prefix
+    s = &s[2..];
+
+    // Read an integer part (if presented)
+    let length = s.chars().take_while(|c| c.is_ascii_hexdigit()).count();
+    let integer = u64::from_str_radix(&s[..length], 16).unwrap_or(0);
+    s = &s[length..];
+
+    // Read a fractional part (if presented)
+    let fractional = if s.starts_with('.') {
+        s = &s[1..];
+        let length = s.chars().take_while(|c| c.is_ascii_hexdigit()).count();
+        let value = parse_fractional_part(&s[..length])?;
+        s = &s[length..];
+        Some(value)
+    } else {
+        None
+    };
+
+    // Read a power (if presented)
+    let power = if s.starts_with('p') || s.starts_with('P') {
+        s = &s[1..];
+        let length = s
+            .chars()
+            .take_while(|c| c.is_ascii_digit() || *c == '-' || *c == '+')
+            .count();
+        let value = s[..length].parse().map_err(|_| ParseNumberError::Float)?;
+        s = &s[length..];
+        Some(value)
+    } else {
+        None
+    };
+
+    // Post-checks:
+    // - Both 'fractional' and 'power' values cannot be 'None' at the same time.
+    // - The entire string must be consumed; otherwise, there could be garbage symbols after the Hex float.
+
+    if fractional.is_none() && power.is_none() {
+        return Err(ParseNumberError::Float);
+    }
+
+    if !s.is_empty() {
+        return Err(ParseNumberError::Float);
+    }
+
+    // Build the result
+    let total =
+        sign * (integer as f64 + fractional.unwrap_or(0.0)) * (2.0_f64).powi(power.unwrap_or(0));
+    Ok(total)
+}
+
+/// Parse the fractional part in hexadecimal notation.
+///
+/// # Errors
+///
+/// This function returns an error if the string is empty or contains characters that are not hex digits.
+fn parse_fractional_part(s: &str) -> Result<f64, ParseNumberError> {
+    if s.is_empty() {
+        return Err(ParseNumberError::Float);
+    }
+
+    let mut multiplier = 1.0 / 16.0;
```

Comment:
please document why 16
and maybe use a variable as it is used 3 time here

Reply:
@sylvestre done